external-crates/move/crates/move-analyzer/src/bin/move-analyzer.rs

```diff
@@ -221,7 +231,22 @@ fn main() {
     eprintln!("Shut down language server '{}'.", exe);
 }
 
-fn on_request(context: &Context, request: &Request) {
+fn on_request(context: &Context, request: &Request, shutdown_request_received: bool) -> bool {
```

Comment:
maybe add a docstring to help explain what the return value signifies now (whether to shutdown after this request)?